src/signHash.c

```diff
@@ -89,6 +98,63 @@ UX_FLOW(ux_signhash_flow,
   &ux_signhash_flow_4_step
 );
 
+void ui_display_sign_hash_flow(void) {
+	// Generate a string for the index.
+	snprintf(ctx->indexStr, sizeof(ctx->indexStr), "with Key #%d?", ctx->keyIndex);
+
+	ux_flow_init(0, ux_signhash_flow, NULL);
+}
+
+#else // HAVE_BAGL
+
+static nbgl_layoutTagValue_t pair;
+static nbgl_layoutTagValueList_t pairList = {0};
+static nbgl_pageInfoLongPress_t infoLongPress;
+
+static void transaction_rejected(void) {
+	do_reject();
+}
+
+static void reject_confirmation(void) {
+	nbgl_useCaseConfirm("Reject transaction?", NULL, "Yes, Reject", "Go back to transaction", transaction_rejected);
+}
+
+// called when long press button on 3rd page is long-touched or when reject footer is touched
+static void review_choice(bool confirm) {
+	if (confirm) {
+		do_approve();
```

Comment:
Optional nitpicking: I would use `transaction_rejected` here to make the code "symmetric", i.e. all reject paths pass around `transaction_rejected`.

Reply:
@troelsfr I don't get what you mean here?
I suppose you were talking about replacing `reject_confirmation()` below by `transaction_rejected()`?
In this case there won't be the confirmation request which can be handy.

Reply:
No, I was talking about that `transaction_rejected` is wrapping `do_approve`. For consistency, I would suggest to either use `do_approve` everywhere or `transaction_rejected` everywhere but avoid mixing the two.

This is nitpicking and completely optional.

Reply:
Sorry - forget my comment. I see now that it says `do_approve` not `do_reject`